packages/flutter_tools/lib/src/macos/cocoapods.dart

```diff
@@ -502,6 +502,17 @@ class CocoaPods {
           );
         }
       }
+    } else if (stdout.contains('unknown ISA `PBXFileSystemSynchronizedRootGroup`')) {
+      // CocoaPods does not work with Xcode 16 since it has not yet been
+      // updated to handled synchronized
+      // groups/folders https://github.com/CocoaPods/CocoaPods/issues/12456
+      _logger.printError(
+        'Error: CocoaPods does not support Xcode 16 synchronized groups. '
+            'To fix your Xcode project, '
+            'see https://github.com/flutter/flutter/issues/156733#issuecomment-2415359014 '
```

Comment:
Awesome, thank you! (I took the liberty of editing the comment since the "package contents" part isn't needed, you can open the workspace in Xcode without that).